instrumentation/apache-dubbo-2.7/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/apachedubbo/v2_7/DubboHeadersGetter.java

```diff
@@ -6,14 +6,31 @@
 package io.opentelemetry.instrumentation.apachedubbo.v2_7;
 
 import io.opentelemetry.context.propagation.TextMapGetter;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.Map;
+import org.apache.dubbo.rpc.RpcInvocation;
 
 enum DubboHeadersGetter implements TextMapGetter<DubboRequest> {
   INSTANCE;
 
   @Override
-  @SuppressWarnings("deprecation") // deprecation for dubbo 3.2.15
+  @SuppressWarnings("unchecked") // unchecked for 2.7.6, 2.7.7
   public Iterable<String> keys(DubboRequest request) {
-    return request.invocation().getAttachments().keySet();
+    RpcInvocation invocation = request.invocation();
+    try {
+      // In 2.7.6, 2.7.7, the StringToObjectMap implementation does not correctly retrieve the
+      // keySet. Therefore, it's advisable to always call getObjectAttachments when it is available.
+      Method getObjectAttachmentsMethod = invocation.getClass().getMethod("getObjectAttachments");
```

Comment:
Our usual pattern for cases like this is to do to reflective lookup only once in the class initializer. Here you can use `RpcInvocation.class` instead of `invocation.getClass()` so you wouldn't need to do the lookup on each invocation. In existing code using method handles is preferred over `java.lang.reflect.Method`.